packages/flutter/test/cupertino/text_form_field_row_restoration_test.dart

```diff
@@ -0,0 +1,251 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+const String text = 'Hello World! How are you? Life is good!';
+const String alternativeText = 'Everything is awesome!!';
+
+void main() {
+  testWidgets('CupertinoTextFormFieldRow restoration', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const CupertinoApp(
+        restorationScopeId: 'app',
+        home: TestWidget(),
```

Comment:
Nit: I think it's kind of hard to read like this, because TestWidget seems so generic here. Maybe just rename it to "RestorableCupertinoTextFormFieldRow" and "useExternal" to "useExternalController"? And/or move the widgets other than the CupertinoTextFormFieldRow out?

I don't know, maybe that makes it more confusing haha. Up to you.